### PR TITLE
WIP: Snyk.io integration

### PR DESCRIPTION
**Pull request by richard-ps-zz**

### CODE REVIEW

File: services/estafette/service.go [hunk 1]
```diff
@@ -104,7 +104,7 @@ func (s *service) CreateBuild(ctx context.Context, build contracts.Build, waitFo
 
 	// inject build stages
 	if hasValidManifest {
-		mft, err = api.InjectSteps(s.config.ManifestPreferences, mft, builderTrack, shortRepoSource, s.supportsBuildStatus(build.RepoSource))
+		mft, err = api.InjectSteps(s.config.ManifestPreferences, s.config.Integrations.SnykIO, mft, builderTrack, shortRepoSource, s.supportsBuildStatus(build.RepoSource))
```
Comment (vitorarins) [inline services/estafette/service.go:107]: I think we should also have a config for choosing whether you want the snyk step or not

File: api/manifestInjector.go [hunk 2]
```diff
@@ -26,6 +26,17 @@ func InjectSteps(preferences *manifest.EstafetteManifestPreferences, mft manifes
 		AutoInjected:   true,
 	}
 
+	snykStep := &manifest.EstafetteStage{
+		Name:           "snyk-scan",
+		ContainerImage: fmt.Sprintf("extensions/snyk-scan:%v", builderTrack),
```
Comment (JubbaSmail) [inline api/manifestInjector.go:31]: the image location is: **eu.gcr.io/travix-com/snyk-scanner**; do we need to push it to **extensions/snyk-scan** ?

Comment (richard-ps) [reply api/manifestInjector.go]: I think so. @vitorarins  Do you know if there is any restriction regarding the location of the extension's image?

Comment (vitorarins) [reply api/manifestInjector.go]: The only restriction is that image should be public, which is not the case with `eu.gcr.io/travix-com/snyk-scanner`.
Don't forget this is an open source project. Let's create `estafette-extension-snyk-scanner` repo in github and publish it to Docker Hub.

Comment (richard-ps) [reply api/manifestInjector.go]: Right. Thanks @vitorarins. I will move the repo and image.

Comment (msharbaji) [reply api/manifestInjector.go]: I am with Vitor in this sense, but actually I'd prefer to be the logic in this repo in golang not by simple bash file

**Comment by JorritSalverda** [comment]
I prefer that we make a more free-form config section for injecting stages before and after the user's stages, where we could move the current ones as well. That keeps it more flexible to future extension and other users of Estafette. This does mean we have to rewrite the injection logic to work from config and support a couple of replacements, like the track (dev, stable).

**Comment by yogeshlonkar** [comment]
Closing PR as Integration already done.

**Pull request closed without merging**